red-hat-data-services/odh-dashboard
# Added Segment.io opt-int logic

- [X] JIRA link(s): https://issues.redhat.com/browse/RHODS-871

## Testing instructions
1. Deploy this PR with the [ODH-Dashboard PR](https://github.com/red-hat-data-services/odh-dashboard/pull/178).
2. Ensure both `segment-key-secret` secret and `rhods-segment-key-config` configmap are deployed in the apps Namespace.
3. Check that the `rhods-segment-key-config` is set to **true**.
4. Open odh dashboard and inspect network, you should find a network call called `/segment-key` with a *write key* enabled.
<img width="1728" alt="odh_dashboard_true" src="https://user-images.githubusercontent.com/16117276/149492355-a12c2ed5-7500-40fa-83ed-5718b46298e3.png">

5. Additionally segment should be tracking the interaction.
6. Now change `rhods-segment-key-config` to **false**.
7. Reload odh dashboard and now `/segment-key` should be empty.
<img width="1728" alt="odh_dashboard_false" src="https://user-images.githubusercontent.com/16117276/149492295-33b06ced-c128-4b2d-9845-7ae9e9ceeb7a.png">
